chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/InvitationManager.java

```diff
@@ -42,163 +47,134 @@
-        // could be multiple invitations for the same email so add the time to
-        // make it unique
-        Date currTime = new Date();
-        dto.setInviteToken(Util.md5(emailAddress + currTime.getTime()));
-
-        return createInvitation(dto);
+        return inviteToAccount(Authority.ROLE_ATL, atlId, emailAddress);
+
     }
 
     @Transactional
     @PreAuthorize("@permissions.hasAccess(T(gov.healthit.chpl.permissions.Permissions).INVITATION, "
             + "T(gov.healthit.chpl.permissions.domains.InvitationDomainPermissions).INVITE_DEVELOPER, #developerId)")
-    public InvitationDTO inviteWithDeveloperAccess(String emailAddress, Long developerId)
+    public UserInvitation inviteWithDeveloperAccess(String emailAddress, Long developerId)
             throws UserCreationException, UserRetrievalException, UserPermissionRetrievalException {
-        InvitationDTO dto = new InvitationDTO();
-        dto.setEmail(emailAddress);
-        dto.setPermissionObjectId(developerId);
-        dto.setPermission(userPermissionDao.getPermissionFromAuthority(Authority.ROLE_DEVELOPER));
-        // could be multiple invitations for the same email so add the time to
-        // make it unique
-        Date currTime = new Date();
-        dto.setInviteToken(Util.md5(emailAddress + currTime.getTime()));
-
-        return createInvitation(dto);
+        return inviteToAccount(Authority.ROLE_DEVELOPER, developerId, emailAddress);
     }
 
-    private InvitationDTO createInvitation(InvitationDTO toCreate)
+    private UserInvitation inviteToAccount(String authority, Long permissionObjectId, String emailAddress)
             throws UserCreationException, UserRetrievalException, UserPermissionRetrievalException {
-        InvitationDTO createdInvitation = null;
-        createdInvitation = invitationDao.create(toCreate);
+        Optional<UserPermission> userPermission = this.userPermissions.stream().filter(perm -> perm.getAuthority().equals(authority)).findAny();
+        if (!userPermission.isPresent()) {
+            throw new UserPermissionRetrievalException("No User Permission was found for authority " + authority);
+        }
+
+        UserInvitation invitation = UserInvitation.builder()
+                .emailAddress(emailAddress)
+                .permissionObjectId(permissionObjectId)
+                .hash(Util.md5(emailAddress + System.currentTimeMillis()))
+                .permission(userPermission.get())
+                .build();
+        LOGGER.info("Creating invitation for " + emailAddress);
+        Long createdInvitationId = invitationDao.create(invitation);
+        LOGGER.info("Created invitation for " + emailAddress + " with ID " + createdInvitationId);
+        UserInvitation createdInvitation = invitationDao.getById(createdInvitationId);
+        LOGGER.info("Emailing user...");
+        invitationEmailer.emailInvitedUser(createdInvitation);
+        LOGGER.info("Emaild " + emailAddress);
```

Comment:
```suggestion
        LOGGER.info("Emailed " + emailAddress);
```

More importantly, do we need all of these LOGGER lines? What are the chances one of these steps will fail? I know they're at `info` level, so if they're useful I'm fine leaving them, it just seems like it might be overkill